### ISSUE TITLE
LocaleMatcher match is not matching locale to its parent locale

### ISSUE THREAD

**Issue by jgoodnoe-css** [formatjs/formatjs#4174]
## Which package?

intl.LocaleMatcher

## Describe the bug

The `match` function of LocaleMatcher does not appear to be mapping a locale to it's parent locale. For example, with the supported locales of `[en, es, es-419]`, if the requested locales are `[es-co]` then `match` will return `es`.  I would expect it to return `es-419`. 

Additionally if the supportedLocales are `[en, es-419]`, and again the requested locales are `[es-co]`. Then `match` will return the defaultLocale.

Can see here that it should be mapped to es-419: https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/parentLocales.json#L108

## To Reproduce

```
import { match } from '@formatjs/intl-localematcher';
const defaultLocale = 'en-US'
const requestedLocales = ['es-co'];
const supportedLocales = ['en', 'es', 'es-419'];
const matchedLocale = match(requestedLocales, supportedLocales, defaultLocale);
// result will be 'es'
```

### Codesandbox URL

[![Edit on Codesandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/pensive-black-l85ng9?fontsize=14&hidenavigation=1&theme=dark)

### Reproducible Steps/Repo

Check out repro steps above or code sandbox

## Expected behavior

Should get `es-419` not `es`

**Comment by longlho** [comment]
cool will take a look

**Comment by jgoodnoe-css** [comment]
Looks like discussion #4105 also raises this issue regarding mapping pt-mz to pt-pt.

**Comment by jgoodnoe-css** [comment]
Don't go stale just yet!

**Comment by longlho** [comment]
this has been fixed in fafbc091a174addc076b9a088d09f37c36b2e3e1
